DIPs/DIP1xxy-mvf.md

```diff
@@ -0,0 +1,322 @@
+
+Stage 2's implementation will require finalizing the [Deprecated Features](https://dlang.org/deprecate.html) page, and an entry in the changelog.
+
+This proposed change may cause existing code to no longer compile, or in some cases, change behavior, but such code is arguably already ambiguous, and it would only benefit users to require them to remove such ambiguity from their code by stating their intent through explicit casts.  Furthermore, this deprecation closes a hole in the type system, making the D programming language just that much more robust for writing safe and reliable code.
+
+### Alternatives considered
+
+#### Change to overload resolution rules
+
+[A pull request](https://github.com/dlang/dmd/pull/1942) was created to resolve [Issue 9999](https://issues.dlang.org/show_bug.cgi?id=9999) by adding a special case to partial ordering resolution:
+
+> In partial ordering resolution, bool parameter is treated as less specialized than other integer types (`byte`, `ubyte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `wchar`, `dchar`).
+
+[It was rejected by Walter Bright](https://issues.dlang.org/show_bug.cgi?id=9999#c3):
+
+>I do not agree with this enhancement. First off, making special cases for partial ordering takes a simple, straightforward idea and turns it into a potential morass of conflicting cases that we'll be stuck with forever. Secondly, the only issue here is whether `1` should be implicitly convertible to `bool`.
+
+Walter identified the fundamental source of the problem to be the implicit conversion of integer literals to `bool`.
+
+A change to the overload resolution rules would still break code relying on the existing overload resolution behavior illustrated in [Example B](#example-b) and [Example C](#example-c), resulting in a deprecation process similar, if not the same, to that of the proposed solution, but would not close the hole in the type system.
+
+#### Disallow implicit conversion of integer literals to `bool`, but only for overloaded functions
+
+It may be possible to fix the bugs documented in [issue 9999](https://issues.dlang.org/show_bug.cgi?id=9999) and [10560](https://issues.dlang.org/show_bug.cgi?id=10560)  by preventing implicit conversion of integer literals to `bool`, but only for overloaded functions.
+
+This was rejected for similar reasons as the former; it complicates the compiler's implementation with special cases that need to be implemented, documented, explained, and understood.
+
+Also, like the former, it would still break code relying on the existing overload resolution behavior illustrated in [Example B](#example-b) and [Example C](#example-c), resulting in a deprecation process similar, if not the same, to that of the proposed solution, but would not close the hole in the type system.
+
+This solution was also not analyzed in enough detail to determine whether it would even scale.
+
+### Breaking changes / deprecation process
```

Comment:
Expected breakage needs to be iterated here, even when it is mentioned elsewhere in the document.